tools/psa/build_tfm.py

```diff
@@ -0,0 +1,181 @@
+VERSION_FILE_PATH = join(ROOT, 'features/FEATURE_PSA/FEATURE_TFM')
+
+dependencies = {
+    "trusted-firmware-m":
+    ['https://git.trustedfirmware.org/trusted-firmware-m.git',
+     'feature-twincpu'],
+    "mbedtls": ['https://github.com/ARMmbed/mbedtls.git',
+                'mbedtls-2.7.9'],
+    "mbed-crypto": ['https://github.com/ARMmbed/mbed-crypto.git',
+                    'mbedcrypto-1.1.0'],
+    "CMSIS_5": ['https://github.com/ARM-software/CMSIS_5.git', '5.5.0'],
+}
+
+def is_git_installed():
+    """
+    Check if git is installed
+    """
+    command = ['git', '--version']
+    try:
+        with open(os.devnull, 'w') as fnull:
+            return subprocess.call(command, stdout=fnull, stderr=fnull)
+    except OSError as e:
+        return e.errno
+
+def is_git_lfs_installed():
+    """
+    Check if git-lfs is installed
+    """
+    command = ['git-lfs', '--version']
+    try:
+        with open(os.devnull, 'w') as fnull:
+            return subprocess.call(command, stdout=fnull, stderr=fnull)
+    except OSError as e:
+        return e.errno
+
+def run_cmd_and_return_output(command):
+    """
+    Run the command in the sytem and return output.
+    Commands are passed as a list of tokens.
+    E.g. The command 'git remote -v' would be passed in as:
+     ['git', 'remote', '-v']
+
+    :param command: System command as a list of tokens
+    """
+    output = ''
+    logger.debug('[Exec] %s', ' '.join(command))
+    try:
+        with open(os.devnull, 'w') as fnull:
+            output = subprocess.check_output(command, stderr=fnull)
+    except subprocess.CalledProcessError as e:
+        logger.error("The command %s failed with return code: %s",
+                     (' '.join(command)), e.returncode)
+        clean_up_cloned_repos()
+    return output
+
+def detect_and_write_tfm_version(tfm_dir):
+    """
+    Identify the version of TF-M and write it to VERSION.txt
+    :param tfm_dir: The filesystem path where TF-M repo is cloned
+    """
+    cmd = ['git', '-C', tfm_dir, 'describe', '--tags',
+           '--abbrev=12', '--dirty', '--always']
+    tfm_version = run_cmd_and_return_output(cmd)
+    logger.info('TF-M version: %s', tfm_version.strip('\n'))
+    if not isdir(VERSION_FILE_PATH):
+        os.makedirs(VERSION_FILE_PATH)
+    with open(join(VERSION_FILE_PATH, 'VERSION.txt'), 'w') as f:
+        f.write(tfm_version)
+
+def check_repo_version(name, deps):
+    """
+    Compare the version of cloned and expected and exit if they dont match
+    :param name: Name of the git repository
+    :param deps: Dictionary containing dependency details
+    """
+    basedir = TF_M_BUILD_DIR
+    if name == 'trusted-firmware-m':
+        cmd = ['git', '-C', join(basedir, name),
+               'rev-parse', '--abbrev-ref', 'HEAD']
+    else:
+        cmd = ['git', '-C', join(basedir, name),
+               'describe', '--tags']
+    _out = run_cmd_and_return_output(cmd)
+    if _out.strip('\n') != deps.get(name)[1]:
```

Comment:
Getting an error with respect executing this script
`mbed-os$ /usr/bin/python3 /mbed-os/tools/psa/build_tfm.py
[TF-M-Builder] 12:44:27: trusted-firmware-m repo exists, checking git version....
Traceback (most recent call last):
  File "/mbed-os/tools/psa/build_tfm.py", line 181, in <module>
    main()
  File "/mbed-os/tools/psa/build_tfm.py", line 173, in main
    clone_tfm_repo()
  File "/mbed-os/tools/psa/build_tfm.py", line 149, in clone_tfm_repo
    check_and_clone_repo('trusted-firmware-m', dependencies)
  File "/mbed-os/tools/psa/build_tfm.py", line 143, in check_and_clone_repo
    check_repo_version(name, deps)
  File "/mbed-os/tools/psa/build_tfm.py", line 119, in check_repo_version
    if _out.strip('\n') != deps.get(name)[1]:
TypeError: a bytes-like object is required, not 'str'`

Is it related to python3, if yes are we not supporting Python 3 as Python 2.7 support will be ending soon?

Reply:
@jainvikas8 Added support for python3